src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -109,16 +115,39 @@ protected void ValidateScopeInput(SortedSet<string> scopesToValidate)
 
         public async Task<AuthenticationResult> RunAsync()
         {
-            AuthenticationResult result = null;
             //this method is the common entrance for all token requests, so it is a good place to put the generic Telemetry logic here
             AuthenticationRequestParameters.RequestContext.TelemetryRequestId = Telemetry.GetInstance().GenerateNewRequestId();
+            var apiEvent = new ApiEvent()
+            {
+                ApiId = ApiId,
+                ValidationStatus = AuthenticationRequestParameters.ValidateAuthority.ToString(),
+                UserId = AuthenticationRequestParameters.User != null ? AuthenticationRequestParameters.User.Identifier : "",
+                CorrelationId = AuthenticationRequestParameters.RequestContext.CorrelationId,
+                RequestId = AuthenticationRequestParameters.RequestContext.TelemetryRequestId,
```

Comment:
I understand your point, but here we don't have much options. Because the left-hand side `RequestId` is a field name of the ApiEvent, defined in the telemetry design document, and it makes sense: in the telemetry context, we do not need an extra "telemetry-" prefix.

At the right-hand side, which is outside of a telemetry event class, I did name it "TelemetryRequestId" for better description. We have the same kind of mindset. :-)